astropy/cosmology/tests/test_parameter.py

```diff
@@ -406,6 +400,67 @@ def func(cosmology, param, value):
         finally:
             param.__class__._registry_validators.pop("newvalidator", None)
 
+    # -------------------------------------------
+
+    def test_Parameter_clone(self, param):
+        """Test :meth:`astropy.cosmology.Parameter.clone`."""
+        # this implicitly relies on `__eq__` testing properly. Which is tested.
+
+        # basic test that nothing changes
+        assert param.clone() == param
+        assert param.clone() is not param  # but it's not a 'singleton'
+
+        # passing kwargs will change stuff
+        newparam = param.clone(unit="km/(yr sr)")
+        assert newparam.unit == u.km / u.yr / u.sr
+        assert param.unit != u.km / u.yr / u.sr  # original is unchanged
+
+        # expected failure for not-an-argument
+        with pytest.raises(TypeError):
+            param.clone(not_a_valid_parameter=True)
+
+    # -------------------------------------------
+
+    def test_Parameter_equality(self):
+        """Test Parameter equality. Determined from the @property."""
+        p1 = Parameter(unit="km / (s Mpc)")
+        p2 = Parameter(unit="km / (s Mpc)")
+        assert p1 == p2
+
+        # not equal parameters
+        p3 = Parameter(unit="km / s")
+        assert p3 != p1
+
+        # misc
+        assert p1 != 2  # show doesn't error
+
+    # -------------------------------------------
+
+    def test_Parameter_repr(self, cosmo_cls, param):
+        """Test Parameter repr."""
+        r = repr(param)
+
+        assert "Parameter(" in r
+        for subs in ("derived=False", 'unit=Unit("m")', 'equivalencies=[(Unit("kg"), Unit("J")',
+                     "fmt='.3g'", "doc='Description of example parameter.'"):
+            assert subs in r, subs
+
+        # fvalidate is a little tricker b/c one of them is custom!
+        if param.fvalidate in param._registry_validators.values():  # not custom
+            assert "fvalidate='default'" in r
+        else:
+            assert "fvalidate='<" in r  # some function
+
+    @pytest.mark.skipif(sys.version_info < (3, 9), reason="unparse needs py3.9+")
+    def test_Parameter_repr_parses(self, param):
+        """Since ``eval`` is unsafe, this just tests parse-ability."""
```

Comment:
I think inside a test `eval` is OK, no? Sort-of nice to see if it actually works!

Reply:
I'm very used to my linter, which loudly complains about ``eval``. I can adjust the test .

Reply:
Done. Test is now called ``test_Parameter_repr_roundtrip``